File: src/Ej3/Razas/Director.java
package Razas;

public class Director{
	private Personaje personaje;
	private PersonajeBuilder personajeBuilder;

	public Director(PersonajeBuilder personajeBuilder){
		this.personajeBuilder = personajeBuilder;
	}

	public void crearPersonaje(String nombre){
		personajeBuilder.CrearPersonaje();
		personaje = personajeBuilder.GetPersonaje();
		personaje.SetName(nombre);
	}

	public void mostrarPersonaje(){
		personaje.mostrarPersonaje();
	}

	public Personaje GetPersonaje(){
		return personaje;
	}

}

File: Practica 2/ej4_creador_personajes/lib/Razas/director.dart

import 'package:ej4_creador_personajes/Razas/Personaje.dart';
import 'package:ej4_creador_personajes/Razas/PersonajeBuilder.dart';

class Director{
  late Personaje personaje;
  late PersonajeBuilder personajeBuilder;

  Director(PersonajeBuilder personajeBuilder){
    this.personajeBuilder = personajeBuilder;
  }

  void crearPersonaje(String nombre){
    personajeBuilder.crearPersonaje();
    personaje = personajeBuilder.getPersonaje();
    personaje.SetName(nombre);
  }

  Personaje getPersonaje(){
    return personaje;
  }

}